src/docfx/build/toc/parser/TableOfContentsParser.cs

```diff
@@ -29,26 +29,32 @@ public static (List<Error> errors, List<TableOfContentsItem> items, JObject meta
             return (errors, items, inputModel?.Metadata, fileMonikers);
         }
 
-        private static (List<Error> errors, TableOfContentsInputModel tocModel) LoadTocModel(Context context, Document file)
+        private static (List<Error> errors, TableOfContentsInputModel tocModel) LoadTocModel(Context context, Document file, string content = null)
         {
             var filePath = file.FilePath;
+
+            if (file.IsFromHistory)
+            {
+                Debug.Assert(!string.IsNullOrEmpty(content));
+            }
+
             if (filePath.EndsWith(".yml", PathUtility.PathComparison))
             {
-                var (errors, tocToken) = YamlUtility.Deserialize(file, context);
+                var (errors, tocToken) = string.IsNullOrEmpty(content) ? YamlUtility.Deserialize(file, context) : YamlUtility.Deserialize(content);
```

Comment:
it should be `content == null`, content read from history might be empty ""

Reply:
yes